### PR TITLE
TEST PULLREQ -- DO NOT MERGE

### PR DESCRIPTION
**Pull request by thatothermitch**

### CODE REVIEW

**Comment by mandrews** [comment]
Is the logger.log method compatible with other common rack /rails loggers?
http://rack.rubyforge.org/doc/classes/Rack/CommonLogger.html
http://clogger.rubyforge.org

Also seems like the format string could be more modular.  Some other common formats:
http://httpd.apache.org/docs/1.3/logs.html#common
http://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/TTCCLayout.html

File: lib/oauth2/provider/rack/access_token_handler.rb [hunk 1]
```diff
@@ -14,6 +16,7 @@ def process
       if request.post?
         block_unsupported_grant_types || handle_basic_auth_header || block_invalid_clients || handle_grant_type
       else
+        log "CLIENT ERROR: token endpoint only supports POST"
```
Comment (mandrews) [inline lib/oauth2/provider/rack/access_token_handler.rb]: seems like this should be something like
 `log 'token endpoint', :level => :error`

Comment (thatothermitch) [inline lib/oauth2/provider/rack/access_token_handler.rb]: Good point, I should update the levels to match.

**Comment by thatothermitch** [comment]
CommonLogger and the Apache Common Log Format are really focused on logging requests. IMO they'd be useful in addition to this type of logging, which is more focused on describing the flow through the underlying business logic. In apache terms this logging would be more like "ScriptLog"

That being said, I would love a format that I could comply with.

File: lib/oauth2/provider/rack/authorization_code_request.rb [hunk 2]
```diff
@@ -1,7 +1,13 @@
 module OAuth2::Provider::Rack
   class AuthorizationCodeRequest
-    def initialize(params)
+    include OAuth2::Provider::Logging
+   
+    attr_reader :env
+
+    def initialize(params, env)
```
Comment (thatothermitch) [inline lib/oauth2/provider/rack/authorization_code_request.rb:7]: Adding a second param here is kind of a fail. A better pattern might be to act like a decorator, i.e.:

```
rack_req = Rack::Request.new(env) # parses env and provides the params method, as well as an env method! -- rails provides controller.request
AuthorizationCodeRequest.new rack_req
```

File: spec/logging_spec.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
+require "spec_helper"
```
Comment (thatothermitch) [inline spec/logging_spec.rb]: this could use some whitespace cleanup

**Pull request closed without merging**